execute/tokendata/processor.go

```diff
@@ -0,0 +1,64 @@
+package tokendata
+
+import (
+	"context"
+
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+
+	"github.com/smartcontractkit/chainlink-ccip/execute/exectypes"
+)
+
+type TokenDataObserver interface {
```

Comment:
Would be nice to have some doc in the interface and maybe the method.

Reply:
I will do as followup, I'm going top-down here, so I wanted to start with interfaces. I think it would clarify once I have an implementation ready, then I will add missing docs